File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/mediator/ScrumMaster.java
package mediator;

class ScrumMaster implements Mediator {
    private Developer developer;
    private Tester tester;

    public void setDeveloper(Developer developer) {
        this.developer = developer;
    }

    public void setTester(Tester tester) {
        this.tester = tester;
    }

    @Override
    public void request(String message, Colleague colleague) {
        if (colleague == developer) {
            tester.receive(message);
        } else if (colleague == tester) {
            developer.receive(message);
        }
    }

    @Override
    public void response(String message, Colleague colleague) {
        System.out.println("Scrum Master handling response: " + message);
    }
}

File: code/csharp/DesignPatterns/mediator/ScrumMaster.cs
namespace Mediator;

class ScrumMaster : IMediator
{
    private Developer _developer;
    private Tester _tester;

    public void SetDeveloper(Developer developer)
    {
        _developer = developer;
    }

    public void SetTester(Tester tester)
    {
        _tester = tester;
    }

    public void Request(string message, Colleague colleague)
    {
        if (colleague == _developer)
        {
            _tester.Receive(message);
        }
        else if (colleague == _tester)
        {
            _developer.Receive(message);
        }
    }

    public void Response(string message, Colleague colleague)
    {
        System.Console.WriteLine($"Scrum Master handling response: {message}");
    }
}
